blockchain/interceptors.go

```diff
@@ -20,79 +22,183 @@ const (
 	EscrowPaymentType = "escrow"
 )
 
-func (p Processor) jobValidationInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo,
-	handler grpc.StreamHandler) error {
+// GrpcStreamContext contains information about gRPC call which is used to
+// validate payment and pricing.
+type GrpcStreamContext struct {
+	MD   metadata.MD
+	Info *grpc.StreamServerInfo
+}
 
-	md, ok := metadata.FromIncomingContext(ss.Context())
-	if !ok {
-		return status.Errorf(codes.InvalidArgument, "missing metadata")
+// Payment represents payment handler specific data which is validated
+// and used to complete payment.
+type Payment interface{}
+
+// PaymentHandler interface which is used by gRPC interceptor to get, validate
+// and complete payment. There are two payment handler implementations so far:
+// jobPaymentHandler and escrowPaymentHandler.
+type PaymentHandler interface {
+	// Payment extracts payment data from gRPC request context.
+	Payment(context *GrpcStreamContext) (payment Payment, err *status.Status)
```

Comment:
"Payment" word is a noun. Does it have sense to use verb for the method name like "GetPayment"?

Reply:
Go coding style recommends not using ```Get``` in such situations :-), see https://golang.org/doc/effective_go.html#Getters